executor/ddl.go

```diff
@@ -38,12 +39,30 @@ type DDLExec struct {
 	done bool
 }
 
+// toErr converts the error to the ErrInfoSchemaChanged when the schema is outdated.
+func (e *DDLExec) toErr(err error) error {
+	if e.ctx.GetSessionVars().StmtCtx.IsDDLJobReady {
+		return err
+	}
+
+	// Before the DDL job is ready，it encouters an error that may be due to the outdated schema information.
+	// Here we distinguish the ErrInfoSchemaChanged error from other errors.
```

Comment:
Add some description to explain why we don't have to check whether the schema is outdated when `e.ctx.GetSessionVars().StmtCtx.IsDDLJobReady` is true.